src/server/tablet_snap.rs

```diff
@@ -234,32 +261,39 @@ async fn recv_snap_files(
     info!("begin to receive tablet snapshot files"; "file" => %path.display());
     fs::create_dir_all(&path)?;
     let _with_io_type = WithIoType::new(context.io_type);
-    loop {
-        let mut chunk = match stream.next().await {
-            Some(Ok(mut c)) if !c.has_message() => c.take_data(),
-            Some(_) => {
-                return Err(box_err!("duplicated metadata"));
-            }
-            None => break,
-        };
+    while let Some(chunk) = stream.next().await {
+        let mut chunk = chunk?;
+        if chunk.has_message() {
+            return Err(box_err!("duplicated metadata"));
+        }
+        let mut data = chunk.take_data();
         // the format of chunk:
         // |--name_len--|--name--|--content--|
-        let len = chunk[0] as usize;
-        let file_name = box_try!(std::str::from_utf8(&chunk[1..len + 1]));
+        let len = data[0] as usize;
+        let file_name = box_try!(std::str::from_utf8(&data[1..len + 1]));
         let p = path.join(file_name);
         let mut f = File::create(&p)?;
-        let mut size = chunk.len() - len - 1;
-        f.write_all(&chunk[len + 1..])?;
+        if chunk.has_encryption_info() {
+            let info = chunk.take_encryption_info();
+            if let Some(mgr) = &key_manager {
+                mgr.insert_file_raw(p.to_str().unwrap(), info)?;
```

Comment:
Should "fun install_tablet" in components/raftstore-v2/src/operation/ready/snapshot.rs be updated too?
Here the p.to_str() is absolute path and once the whole tablet moved from snapshot folder to tablet folder,  the keys dict should be updated as well.

Reply:
Will be handled in https://github.com/tikv/tikv/pull/14095.